fake-ipa/fake_ipa/inspector.py

```diff
@@ -30,7 +30,7 @@ def inspect(system, inspection_callback_url, verify, cert, logger):
                     "lldp": None,
                     "product": "0x0001",
                     "vendor": "0x1af4",
-                    "name": "eth1",
+                    "name": system.get("nics")[0]["name"] or "enp1s0",
```

Comment:
```suggestion
                    "name": system.get("nics")[0]["name"] or "enp1s0",
```
This will fail with KeyError or ValueError if `.get("nics")` does not exist, is empty, or the first item does not have "name" key. "or" catches only the case the name is empty/None.

Reply:
done ! If nics does not exist then the inspection fail in it get cached in https://github.com/metal3-io/utility-images/blob/f5f03b6b80a6bfbb135ac9ae7324539874c018a1/fake-ipa/fake_ipa/fake_agent.py#L72 

The node has to have at least one interface and mac address

Reply:
OK, the code is now better and also if something else goes haywire it is caiught by the broad exception as linked and insepction will fail. Gotcha.